### PR TITLE
Issue #267: Make each run command stand alone

### PR DESCRIPTION
**Pull request by joshua-rutherford**
@Cheney11M and @justincely, this update actually runs the tests.  

@justincely, I'm not hard over on things being on separate lines but we shouldn't remove the tests. In this case I had to install `dep`, intialize and download the dependencies to get the tests to run.  We should eventually cache the dependencies and such but this gets us back to where we were in 1.0.

@dougfort and @alecholmez, can one of you review this?

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,49 +1,37 @@
...
-          mkdir -p ~/.go_workspace/src/$IMPORT_PATH
-          rm -rf ~/.go_workspace/src/$IMPORT_PATH
-    # - run: cd ~/.go_workspace/src/$IMPORT_PATH && GOPATH=~/.go_workspace go test $(go list ./... | grep -v '/vendor/')
+    - run: mkdir -p download
+    - run: test -e download/$GODIST || curl -o download/$GODIST https://storage.googleapis.com/golang/$GODIST
+    - run: sudo rm -rf /usr/local/go
+    - run: sudo tar -C /usr/local -xzf download/$GODIST
+    - run: curl -OL https://github.com/google/protobuf/releases/download/v3.3.0/$PROTOCDIST
+    - run: sudo rm -rf /usr/bin/protoc
+    - run: sudo unzip $PROTOCDIST -d protoc3
```
Comment (dougfort) [inline .circleci/config.yml]: fwiw there is an include part here that should be copied or symlinked to /usr/include/google/protobuf.

Comment (joshua-rutherford) [reply .circleci/config.yml]: Are you suggesting we add:
```
      - run: sudo ln -s ${PWD}/protoc3/include/protobug /usr/include/google/protobuf
```
It's not in the original config so I didn't add it, but if you tell me what you want linked we can do that.

Comment (dougfort) [reply .circleci/config.yml]: yes, @LutherLu1  sent me this recently. There are bogus versions of google/protobuf floating around. We may not have hit it in CircleCI yet, but that include is what we want.

Comment (dougfort) [reply .circleci/config.yml]: https://github.com/golang/protobuf/issues/642

**Pull request merged**